### DISCUSSION TITLE
Eloquent model changes get lost when typing params in event listeners

### DISCUSSION THREAD

**Discussion by kiessan** [livewire/livewire#4818] [1. Help]
Hi all, I am fairly new to livewire and observed a suspicious behavior when typing params in event listeners.

Please consider the following example:

```php
class GeneralInformation extends Component
{
    public Booking $booking;

    protected array $rules = [
        'booking.invoiceNumber' => ['required', 'min:2'],
    ];

    protected  $listeners = [
        'invoice-number-updated' => 'invoiceNumberUpdated'
    ];

    public function mount(): void
    {
        // invoiceNumber = ABC
        $this->booking = Booking::find(1);
    }

    public function updated($propertyName)
    {
        if($propertyName === 'booking.invoiceNumber') {
            // invoiceNumber = DEF
            $this->emit('invoice-number-updated', $this->booking);
        }
    }

    public function invoiceNumberUpdated(Booking $booking)
    {
        // invoiceNumber = ABC
        dump($booking);
    }

    public function render(): Factory|View|Application
    {
        return view('livewire.general-information');
    }
}
```

* When dumping the `invoiceNumber` in the `updated` method, it it still "DEF"
* When dumping the `invoiceNumber` in the listener, it restored to "ABC"
* When I remove the typing from the listener (` invoiceNumberUpdated(booking)`), the `invoiceNumber` is correctly "DEF".

This doesn't seem to be a desired behavior for.

It would be great if somebody could shed some light on this. 

Thx a lot!

**Comment by silnarm** [comment]
Fascinating little issue.
I can't tell you everything you might want to know sorry, but in broad terms what is happening is that (even though it all happens on the server side in one request processing) the argument for the emit is getting dehydrated (to an array) and hydrated again (but still as an array) before hitting your listener, without the type hint, you get the array, when the type hint is there, Livewire sees this and that the incoming arg is not the same type, so invokes the app container to resolve it using the array. The container is sees the 'id' key in the array and dutifully retrieves the associated model from the database and that gets injected it into your handler.
The whole dehydration/rehydration thing I can't tell you much about, whilst poking around in there I finally decided I was out of my depth and too far in the rabbit warren and in danger on not being able to find my way out ;)

**Comment by joshhanley** [comment]
@kiessan can you share the code where you dispatch the `invoice-number-updated` event? Thanks!

**Reply by kiessan** [reply]
@joshhanley There was a typo in my example. The event is emitted in the `updated()` method.

**Comment by joshhanley** [comment]
@kiessan I wouldn't be calling `emit` there, as it will trigger another request for it to all run properly (as `emit` is dispatched from javascript when the first request is finished).

I would just call your `invoiceNumberUpdated` method directly like:

```php
    public function updated($propertyName)
    {
        if($propertyName === 'booking.invoiceNumber') {
            // invoiceNumber = DEF
            $this->invoiceNumberUpdated($this->booking);
        }
    }
```

Hope this helps!
